types/apostrophe/index.d.ts

```diff
@@ -7,14 +7,24 @@
 export = apostrophe;
 export as namespace apos;
 
-declare function apostrophe(options: any, ...args: any[]): any;
+declare function apostrophe(
+    options: apostrophe.AposConstructor,
+    ...args: any[]
+): any;
 
 declare namespace apostrophe {
     const moogBundle: {
         directory: string;
         modules: string[];
     };
 
+    // Pass in custom modules as first argument
+    // second argument is additional custom options e.g. restApi exposed by apostrophe-headless
+    interface AposConstructor<M = {}, O = {}> {
```

Comment:
This [seems like it should have more members than this](https://github.com/apostrophecms/apostrophe/blob/master/index.js#L289) and that those members should all be optional. Since you're changing it from `any`, having as many of the members people may have been using as possible it good, since it'll make it less likely we'll break them by making this more typesafe.

Reply:
Though tbf, you're the definition owner; so if you say this is fine, I'll believe you.

Reply:
@weswigham PR has been updated.